pandas/core/arrays/period.py

```diff
@@ -844,7 +753,7 @@ def _add_delta_td(self, other):
         #  delta_to_nanoseconds(delta).  Because delta here is an integer,
         #  delta_to_nanoseconds will return it unchanged.
         ordinals = super(PeriodArray, self)._add_delta_td(delta)
-        return type(self)(ordinals, self.freq)
+        return ordinals
 
     def _add_delta_tdi(self, other):
```

Comment:
The superclass version seems to return integers, but this an actual PeriodArray?

Can you add a comment indicating the types?

Reply:
> The superclass version seems to return integers, but this an actual PeriodArray?

Both this and the superclass version return i8 array.  The superclass has a brief docstring to that effect.

Reply:
Can you then add a comment on this function indicating the types? That really makes it easier to read through this code

But, to my original question: this function uses `_addsub_int_array`, which is defined a little bit above to return a PeriodArray, and not a i8 array. 
I assume in practice this is not really a problem, because the code that converts the i8 array to PeriodArray, will also accept a PeriodArray.

Reply:
Good catch, I must have lost track of this somewhere in the rebase process.  Fixed to correctly return an i8 array, and brief docstrings added indicating types.